omnidocs/tasks/math_expression_extraction/extractors/surya_math.py

```
@@ -0,0 +1,189 @@
+
+    def _check_dependencies(self):
+        """Check if required dependencies are available."""
+        try:
+            import surya
+            if self.show_log:
+                logger.info(f"Found surya package at: {surya.__file__}")
+        except ImportError:
+            raise ImportError(
+                "surya-ocr package not found. Please install with: "
+                "pip install surya-ocr"
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)